@xen-orchestra/web/src/components/pool/connect/ConnectionCard.vue

```diff
@@ -0,0 +1,164 @@
+            </UiInfo>
+          </VtsInputWrapper>
+          <!-- TODO validation -->
+          <VtsInputWrapper :label="t('password')">
+            <UiInput v-model="form.password" accent="brand" required type="password" />
+          </VtsInputWrapper>
+        </div>
+      </div>
+      <UiTitle>{{ t('option') }}</UiTitle>
+      <div class="checkbox-wrapper" :class="{ mobile: uiStore.isMobile }">
+        <UiCheckbox v-model="form.readOnly" accent="brand">{{ t('read-only') }}</UiCheckbox>
+        <UiCheckbox v-model="form.allowUnauthorized" accent="brand">
+          {{ t('accept-self-signed-certificates') }}
+        </UiCheckbox>
+      </div>
+      <div class="button-warper">
+        <RouterLink to="/">
+          <UiButton accent="brand" size="medium" variant="secondary">{{ t('cancel') }}</UiButton>
+        </RouterLink>
+        <UiButton type="submit" accent="brand" size="medium" variant="primary" :busy="connecting">
+          {{ t('connect') }}
+        </UiButton>
+      </div>
+    </form>
+  </UiCard>
+</template>
+
+<script setup lang="ts">
+import createAndConnectServer, { type ConnectServerPayload } from '@/jobs/create-server.job'
+import VtsInputWrapper from '@core/components/input-wrapper/VtsInputWrapper.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiCard from '@core/components/ui/card/UiCard.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import UiInput from '@core/components/ui/input/UiInput.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import { useUiStore } from '@core/stores/ui.store'
+import { reactive, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useRouter } from 'vue-router'
+
+const { t } = useI18n()
+const router = useRouter()
+const uiStore = useUiStore()
+const connecting = ref(false)
+
+const form: ConnectServerPayload = reactive({
+  host: history?.state?.ip ?? '',
+  httpProxy: '',
+  username: '',
+  password: '',
+  readOnly: false,
+  allowUnauthorized: false,
+})
+
+function submit() {
+  // Clone the form to avoid reactivity issues
+  const payload = { ...form }
+  connecting.value = true
+  createAndConnectServer(payload)
+    .then(response => {
+      if (response) {
+        router.push({
+          path: '/pool/connect/success',
+          state: {
+            ip: form.host,
+            idServer: response,
+          },
+        })
+      } else {
+        router.push({
+          path: '/pool/connect/error',
+          state: {
+            ip: form.host,
+            errorJson: response,
+          },
+        })
+      }
+    })
+    .catch(reson => {
+      router.push({
+        path: '/pool/connect/error',
+        state: {
+          ip: form.host,
+          ErrorCode: reson.status,
+          errorJson: reson.message,
+        },
+      })
+    })
+    .finally(() => (connecting.value = false))
+}
```

Comment:
I think we can simplify this
```suggestion
async function submit() {
  connecting.value = true
  try {
    const serverId = await createAndConnectServer(form)
    await router.push({
      path: '/pool/connect/success',
      state: { ip: form.host, idServer: serverId },
    })
  } catch (error: any) {
    await router.push({
      path: '/pool/connect/error',
      state: {
        ip: form.host,
        errorCode: error.status,
        errorJson: error.message,
      },
    })
  } finally {
    connecting.value = false
  }
}
```